github/pkg/mtadapter/router.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mtadapter
+
+import (
+	"net/http"
+	"sync"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"go.uber.org/zap"
+	"knative.dev/eventing-contrib/github/pkg/client/listers/sources/v1alpha1"
+)
+
+type keyedHandler struct {
+	handler   http.Handler
+	namespace string
+	name      string
+}
+
+// Router holds the main GitHub webhook HTTP router and delegate to sub-routers
+type Router struct {
```

Comment:
Any reason why using your own router more than using the std lib `ServeMux` or a library like `gorilla`?

Reply:
no particular reason. Changing... thanks!

Reply:
Actually I'll do the change in a subsequent PR. @slinkydeveloper .